presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -346,29 +354,27 @@ public void moreMemoryAvailable()
 
     public void transferMemoryToTaskContext(long taskBytes)
     {
-        long bytes = memoryReservation.getAndSet(0);
-        driverContext.transferMemoryToTaskContext(bytes);
-
-        TaskContext taskContext = driverContext.getPipelineContext().getTaskContext();
-        if (taskBytes > bytes) {
-            try {
-                taskContext.reserveMemory(taskBytes - bytes);
-            }
-            catch (ExceededMemoryLimitException e) {
-                taskContext.freeMemory(bytes);
-                throw e;
-            }
+        long bytes = operatorMemoryContext.getUserMemory();
+        LocalMemoryContext taskLocalMemoryContext = driverContext.getPipelineContext().getTaskContext().localUserMemoryContext();
+        try {
+            operatorMemoryContext.localUserMemoryContext().transferOwnership(taskLocalMemoryContext);
```

Comment:
This explains why you call get parent inside `transferOwnership`. You only exposed `localUserMemoryContext`, not `aggregatedUserMemoryContext`. As a result, you don't have access to it's parent.

This solution is likely not ok. I'll revisit．